modules/caddyhttp/fileserver/matcher.go

```diff
@@ -179,19 +182,20 @@ func (m MatchFile) selectFile(r *http.Request) (matched bool) {
 	}
 
 	// common preparation of the file into parts
-	prepareFilePath := func(file string) (string, string) {
-		suffix := m.firstSplit(path.Clean(repl.ReplaceAll(file, "")))
+	prepareFilePath := func(file string) (string, string, string) {
```

Comment:
```suggestion
	prepareFilePath := func(file string) (suffix, fullpath, remainder string) {
```

Named returns here might be a good idea since it keeps the return values unambiguous. Note that the following lines will need to be updated to not use the `:=` notation.